mipengine/controller/algorithms_specifications.py

```diff
@@ -81,22 +83,28 @@ def __init__(self):
         algorithms_path = Path(algorithms.__file__).parent
 
         all_algorithms = {}
-        for algorithm_property_path in algorithms_path.glob('*.json'):
+        for algorithm_property_path in algorithms_path.glob("*.json"):
             try:
-                algorithm = AlgorithmSpecifications.from_json(open(algorithm_property_path).read())
+                algorithm = AlgorithmSpecifications.from_json(
+                    open(algorithm_property_path).read()
+                )
             except Exception as e:
                 logging.error(f"Parsing property file: {algorithm_property_path}")
                 raise e
             all_algorithms[algorithm.name] = algorithm
 
+        # The algorithm key should be in snake case format, to make searching for an algorithm easier.
         self.enabled_algorithms = {
-            algorithm.name: algorithm
+            str.lower(algorithm.name).replace(" ", "_"): algorithm
```

Comment:
Why are you replacing " "  with "_" here?
Also you can algorithm.name.lower() instead of str.lower(algorithm.name) to be more clear.

Reply:
Good point, I will make a fix.